### PR TITLE
The Very Popular Jar Update

### PR DESCRIPTION
**Pull request by InsightfulParasite**
## About The Pull Request
Partially inspired by Shiptest beakers being sealable. Jars now can have their tops taken on and off.

Additionally adds a recipe for jars but will remove if it seems too much for this PR.

## Why It's Good For The Game
Fixes jars being glass containers filled with deliciousness.

Makes jars more common.

## Changelog
:cl:
add: mildly expensive jar crafting recipie
fix: jars being undrinkable.
tweak: jars now can be alt clicked open and closed.
/:cl:

### CODE REVIEW

**Comment by Gboster-0** [comment]
please replace the \<span class='notice'></span> thingies with span classes
here's a regex ya can use:
```
"<span class='(.*?)'>(.*?)<\/span>"
span_$1("$2")
```

File: code/modules/reagents/reagent_containers/glass.dm [hunk 1]
```diff
@@ -145,15 +145,104 @@
...
+/obj/item/reagent_containers/glass/beaker/jar/afterattack(obj/target, mob/user, proximity)
+	if(cap_on && (target.is_refillable() || target.is_drainable() || (reagents.total_volume && user.a_intent == INTENT_HARM)))
+		to_chat(user, "<span class='warning'>You must remove the cap before you can do that!</span>")
+		return
+
+	else if(istype(target, /obj/item/reagent_containers/glass/beaker/jar))
+		var/obj/item/reagent_containers/glass/beaker/jar/WB = target
+		if(WB.cap_on)
+			to_chat(user, "<span class='warning'>[WB] has a cap firmly twisted on!</span>")
+	. = ..()
```
Comment (Gboster-0) [inline code/modules/reagents/reagent_containers/glass.dm:245]:
```suggestion
	return ..()
```

File: code/modules/reagents/reagent_containers/glass.dm [hunk 2]
```diff
@@ -145,15 +145,104 @@
...
+
+/obj/item/reagent_containers/glass/beaker/jar/is_refillable()
+	if(cap_on)
+		return FALSE
+	. = ..()
+
+/obj/item/reagent_containers/glass/beaker/jar/is_drainable()
+	if(cap_on)
+		return FALSE
+	. = ..()
```
Comment (Gboster-0) [inline code/modules/reagents/reagent_containers/glass.dm:220]:
```suggestion
	return ..()
```

File: code/modules/reagents/reagent_containers/glass.dm [hunk 3]
```diff
@@ -145,15 +145,104 @@
...
+	else
+		cap_on = TRUE
+		spillable = FALSE
+		to_chat(user, "<span class='notice'>You put the cap on [src].</span>")
+	update_icon()
+
+/obj/item/reagent_containers/glass/beaker/jar/is_refillable()
+	if(cap_on)
+		return FALSE
+	. = ..()
```
Comment (Gboster-0) [inline code/modules/reagents/reagent_containers/glass.dm:215]:
```suggestion
	return ..()
```

**Review by Gboster-0 — COMMENTED** [review]
also quite sure those should be `return ..()`

**Comment by InsightfulParasite** [comment]
> please replace the <span class='notice'></span> thingies with span classes here's a regex ya can use:
> 
> ```
> "<span class='(.*?)'>(.*?)<\/span>"
> span_$1("$2")
> ```

I completely forgot about this.

**Comment by InsightfulParasite** [comment]
Should be good now. Had to do a quick test to make sure beakers didnt explode from runtimes.

**Comment by Coxswain-Navigator** [comment]
Can you stick a plushie in them though? If not, might as well port jars from /vg/ then.

**Comment by InsightfulParasite** [comment]
> Can you stick a plushie in them though? If not, might as well port jars from /vg/ then.

What does /vg/ jars do?

**Review by Kitsunemitsu — APPROVED** [review]
Mostly a span update.

**Pull request merged**